pkg/queues/redis_provider.go

```diff
@@ -483,6 +482,30 @@ type redisQueue struct {
 	retryConfig  RetryConfig
 }
 
+// ensureConsumerGroup recreates the consumer group if it was lost (e.g., after Redis restart).
+func (r *redisQueue) ensureConsumerGroup(ctx context.Context) error {
+	// This is idempotent - no need to synchronize
+	err := r.client.XGroupCreateMkStream(ctx, r.name, r.groupName, "0").Err()
+	if err == nil {
```

Comment:
can we try to align with the standard golang error handling, eval err !=nil first and return early. it makes error handling more predictable and easier to maintain. 

```
	if err := r.client.XGroupCreateMkStream(ctx, r.name, r.groupName, "0").Err(); err != nil {
		if allowExists && strings.Contains(err.Error(), "BUSYGROUP") {
			return nil
		}
		return fmt.Errorf("failed to create consumer group: %w", err)
	}
	[...]
```